packages/components/addon/components/hds/form/character-count/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+const ID_PREFIX = 'character-count-';
+const NOOP = () => {};
+
+export default class HdsFormCharacterCountIndexComponent extends Component {
+  // The current number of characters in the associated input
+  @tracked currentLength;
+  inputControl = document.getElementById(this.args.controlId);
+
+  /**
+   * @param maxLength
+   * @type {number}
+   * @default null
+   * @description The maximum number of characters allowed.
+   */
+  get maxLength() {
+    let { maxLength } = this.args;
+    return maxLength;
+  }
+
+  /**
+   * @param remainingLength
+   * @type {number}
+   * @default null
+   * @description The remaining number of characters.
+   */
+  get remainingLength() {
+    return this.maxLength - this.currentLength;
+  }
+
+  /**
+   * Determines the unique ID to assign to the element
+   * @method id
+   * @return {(string|null)} The "id" attribute to apply to the element or null, if no controlId is provided
+   */
+  get id() {
+    let { controlId } = this.args;
+    if (controlId) {
+      return `${ID_PREFIX}${controlId}`;
+    }
+    return null;
+  }
+
+  /**
+   * @param onInsert
+   * @type {function}
+   * @default () => {}
+   */
+  get onInsert() {
+    let { onInsert } = this.args;
+
+    if (!this.inputControl || !this.inputControl.value) {
+      console.error(
+        '`Hds::Form::CharacterCount` component - `@controlId` selector provided does not point to a valid input element, check the id',
+        this.args.controlId
+      );
+    } else {
+      this.updateCurrentLength();
+      this.inputControl.addEventListener(
+        'input',
+        this.updateCurrentLength,
+        true
+      );
+    }
+
+    // notice: this is a guard used to prevent triggering an error when the component is used as standalone element
+    if (typeof onInsert === 'function') {
+      return onInsert;
+    } else {
+      return NOOP;
+    }
+  }
+
+  @action
+  willDestroyNode() {
+    if (this.inputControl) {
+      this.inputControl.removeEventListener(
+        'input',
+        this.updateCurrentLength,
+        true
+      );
+    }
+  }
+
+  @action
+  updateCurrentLength() {
```

Comment:
do you remember we talked about having a callback function\* that the consumers can hook into? do you think it still makes sense to have it? otherwise how consumers would know when to trigger/show the error?

\* (with the function returning all the relevant values: `currentLenght`,`maxLength`, `remainingLength`)

Reply:
as requested, will add these in a new commit